code/__HELPERS/game.dm

```diff
@@ -214,17 +214,21 @@
 /proc/get_hearers_in_view(R, atom/source)
 	// Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode.
 	var/turf/T = get_turf(source)
-	var/list/hear = list()
+	. = list()
 
 	if(!T)
-		return hear
-
-	var/list/range = get_hear(R, T)
-	for(var/atom/movable/A in range)
-		hear |= recursive_hear_check(A)
-
-	return hear
+		return
 
+	if (R == 0) // if the range is zero, we know exactly where to look for, we can skip view
+		. = recursive_hear_check(T)
+	else  // A variation of get_hear inlined here to take advantage of the compiler's fastpath for obj/mob in view
+		var/lum = T.luminosity
+		T.luminosity = 6
+		for(var/obj/O in view(R, T))
+			. += recursive_hear_check(O)
+		for(var/mob/M in view(R, T))
+			. += recursive_hear_check(M)

```

Comment:
recursive_hear_check takes a list, you could likely speed this up even more by saving this to a temp list and calling recursive_hear_check once.

Reply:
i will try tomorrow